openwisp_firmware_upgrader/routing.py

```diff
@@ -0,0 +1,14 @@
+from django.urls import re_path
+
+from .websockets import BatchUpgradeProgressConsumer, UpgradeProgressConsumer
+
+websocket_urlpatterns = [
+    re_path(
+        r"ws/upgrade/(?P<operation_id>[^/]+)/$",
+        UpgradeProgressConsumer.as_asgi(),
+    ),
+    re_path(
+        r"ws/batch-upgrade/(?P<batch_id>[^/]+)/$",
+        BatchUpgradeProgressConsumer.as_asgi(),
+    ),
+]
```

Comment:
I would suggest constructing these URL patterns similar to how we do in other modules - for example:
https://github.com/openwisp/openwisp-controller/blob/85eee35ea52c64018c64baa9d6e61507ffdd4831/openwisp_controller/connection/channels/routing.py#L1-L12

Also, let's follow a consistent directory structure:
`channels/routing.py`, `channels/consumer.py`, and `channels/producer.py`, just like we do elsewhere. :pray:

Reply:
Also, don’t forget to add these new routes to the main ASGI application: https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/tests/openwisp2/routing.py